convert2rhel/subscription.py

```diff
@@ -223,18 +223,27 @@ def remove_original_subscription_manager():
 
 
 def install_rhel_subscription_manager():
-    loggerinst.info("Installing subscription-manager RPMs.")
+    loggerinst.info("Checking for subscription-manager RPMs.")
     rpms_to_install = [os.path.join(SUBMGR_RPMS_DIR, filename) for filename in os.listdir(SUBMGR_RPMS_DIR)]
 
     if not rpms_to_install:
-        loggerinst.warn("No RPMs found in %s." % SUBMGR_RPMS_DIR)
+        loggerinst.warning("No RPMs found in %s." % SUBMGR_RPMS_DIR)
         return
 
+    # These commands have to be here, before the call for installing subscription-manager this is intended, otherwise
+    # `pkghandler.filter_installed_pkgs()` would return every single package that is listed in `rpms_to_install` and
+    # we don't want this to happen. We want only the packages that were here before the instalation of
+    # subscription-manager.
+    pkg_names = get_installed_submgr_pkg_names(rpms_to_install)
+    rpms_to_untrack = pkghandler.filter_installed_pkgs(pkg_names)
```

Comment:
```suggestion
    # These functions have to be called before installation of the subscription-manager packages, otherwise
    # `pkghandler.filter_installed_pkgs()` would return every single package that is listed in `rpms_to_install` and
    # we don't want this to happen. We want to know about the packages that were installed before the installation of
    # subscription-manager.
    pkg_names = get_pkg_names_from_rpm_paths(rpms_to_install)
    pkgs_to_not_track = pkghandler.filter_installed_pkgs(pkg_names)
```